lib/util.js

```diff
@@ -433,3 +433,38 @@ function parseSourceMapInput(str) {
   return JSON.parse(str.replace(/^\)]}'[^\n]*\n/, ''));
 }
 exports.parseSourceMapInput = parseSourceMapInput;
+
+/**
+ * Compute the URL of a source given the the source root, the source's
+ * URL, and the source map's URL.
+ */
+function computeSourceURL(sourceRoot, sourceURL, sourceMapURL) {
+  sourceURL = sourceURL || '';
+
+  if (sourceRoot) {
+    // This follows what Chrome does.
+    if (sourceRoot[sourceRoot.length - 1] !== '/' && sourceURL[0] !== '/') {
+      sourceRoot += '/';
+    }
+    // The source map spec specifies plain prepending here.
+    sourceURL = sourceRoot + sourceURL;
+  }
+
+  if (sourceMapURL) {
```

Comment:
I would add a few more comments here explaining why you are doing this work, and how it conforms to the spec. It feels a bit mysterious to me.